### PR TITLE
Chore: Bump Go to 1.23.0

### PR DESCRIPTION
**Pull request by hairyhenderson**
**What is this feature?**

Bump to Go 1.23.0. Includes necessary upgrade to golangci-lint as well.

### CODE REVIEW

**Comment by hairyhenderson** [comment]
Sorry for the mass notifications - lots of lint errors to fix across the codebase 😓

File: pkg/build/wire/internal/wire/testdata/UnexportedStruct/want/wire_errs.txt [hunk 1]
```diff
@@ -1 +1 @@
-example.com/foo/wire.go:x:y: foo not exported by package bar
\ No newline at end of file
+example.com/foo/wire.go:x:y: name foo not exported by package bar
```
Comment (ryantxu) [inline pkg/build/wire/internal/wire/testdata/UnexportedStruct/want/wire_errs.txt:1]: I added this blindly based on https://drone.grafana.net/grafana/grafana/192278/5/7

**Comment by ryantxu** [comment]
@hairyhenderson -- I fixed the failing tests.  I understand 2/3; 

For the wire snapshot failure -- i just updated the snapshot.  Seems safe, but honestly I have no idea 🤷🏻

**Comment by hairyhenderson** [comment]
thanks @ryantxu, I appreciate it!

File: pkg/services/annotations/annotationsimpl/composite_store.go [hunk 2]
```diff
@@ -2,12 +2,14 @@ package annotationsimpl
...
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sort"
 
 	"github.com/grafana/grafana/pkg/infra/log"
 
 	"github.com/grafana/dskit/concurrency"
+
```
Comment (mildwonkey) [inline pkg/services/annotations/annotationsimpl/composite_store.go:12]: ?

Comment (hairyhenderson) [reply pkg/services/annotations/annotationsimpl/composite_store.go:12]: 🤦‍♂️ thanks - I meant to keep an eye on the goimports shenanigans!

**Comment by hairyhenderson** [comment]
(rebased so the Drone build would work)

**Review by baldm0mma — APPROVED** [review]

**Pull request merged**